### PR TITLE
add HW_10 dir

### PR DESCRIPTION
**Pull request by dymekkqa**

### CODE REVIEW

File: HW_10/task_1.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    getDetails(): string;
+}
+
+abstract class Employee implements IPerson {
+    protected salary = 0;
+    constructor(
+        public name: string,
+        public surname: string,
+        public experienceOfYears: number
+    ) {};
```
Comment (danikkrapivnitskiy) [inline HW_10/task_1.ts:21]: Add there calculateSalary method and you won't be needed to add it the all child classes

File: HW_10/task_1.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+type TMethodology = 'Scrum' | 'Kanban';
+
+class Manager extends Employee{
+    constructor(
+        name: string,
+        surname: string,
+        experienceOfYears: number,
+        public prefered: TMethodology
```
Comment (danikkrapivnitskiy) [inline HW_10/task_1.ts:34]: recommend to use private modificator if you don't use the field outside the class

File: HW_10/task_1.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+type TProgrammingLanguages = 'JS' | 'Java' | 'TS' | 'Python'
+
+class Developer extends Employee{
+    constructor(
+        name: string,
+        surname: string,
+        experienceOfYears: number,
+        public programmingLanguage: TProgrammingLanguages
```
Comment (danikkrapivnitskiy) [inline HW_10/task_1.ts:54]: same

File: HW_10/task_2.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+function map(arr, callback) {
+    const result = arr.map(function (currentValue, index) {
+        return callback(currentValue, index, arr);
+    });
+    return result;
+}
+const numbers = [1, 2, 3, 4, 5];
+const doubledNumbers = map(numbers, (num, index) => num * index);
+console.log(doubledNumbers);
+function generateObject(arr) {
```
Comment (danikkrapivnitskiy) [inline HW_10/task_2.js:11]: function without types

File: HW_10/task_2.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    const obj = {};
+    for (let i = 0; i < arr.length; i++) {
```
Comment (danikkrapivnitskiy) [inline HW_10/task_2.js:13]: recommend to use Object.fromEntries

File: HW_10/task_2.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+const numbers = [1, 2, 3, 4, 5];
+const doubledNumbers = map(numbers, (num, index) => num * index);
+console.log(doubledNumbers)
+
+
+
+function generateObject<T>(arr: [string, T][]): {[key: string]: T}{
+    const obj : {[key: string]: T} = {}
+    for (let i = 0; i < arr.length; i++){
```
Comment (danikkrapivnitskiy) [inline HW_10/task_2.ts:18]: recommend to use Object.fromEntries

File: HW_10/task_2.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    return result
+}
+
+const numbers = [1, 2, 3, 4, 5];
+const doubledNumbers = map(numbers, (num, index) => num * index);
+console.log(doubledNumbers)
+
+
+
+function generateObject<T>(arr: [string, T][]): {[key: string]: T}{
```
Comment (danikkrapivnitskiy) [inline HW_10/task_2.ts:16]: better to create type as prevention of duplication

File: HW_10/task_2.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+type TCallback<T, U> = (currentValue: T, index: number, arr: T[]) => U
+
+function map<T, U>(arr: T[], callback: TCallback<T, U>): U[] {
+    const result = arr.map(function(currentValue, index) {
+        return callback(currentValue, index, arr);
```
Comment (anatoly-karpovich) [inline HW_10/task_2.ts:5]:
```suggestion
    const result = arr.map(callback);
```
I was waiting for a for loop, cheater)

**Review by anatoly-karpovich — APPROVED** [review]

**Pull request merged**